cloud-functions/upload_fhir_bundle/main.py

```diff
@@ -46,30 +47,29 @@ def upload_fhir_bundle(request: flask.Request) -> flask.Response:
     :return: Returns a flask.Response object containing and overall response from the
         FHIR store as well as for the upload of each individual resource in the bundle.
     """
+
+    content_type = "application/json"
     # Validate request header.
-    if request.headers.get("Content-Type") != "application/json":
-        logging.error("Header must inclue: 'Content-Type:application/json'.")
-        return {
-            "status": 400,
-            "summary": "Bad request",
-            "description": "Header must inclue: 'Content-Type:application/json'.",
-        }
-    request_json = request.get_json(silent=False)
+    header_response = validate_request_header(request, content_type)
+
+    if header_response.status_code == 400:
+        return header_response
 
+    request_json = request.get_json(silent=False)
     # Validate request body.
     try:
         request_body = RequestBody.parse_obj(request_json)
     except ValidationError as error:
         logging.error(error)
         error_as_dictionary = json.loads(error.json())[0]
-        return {
-            "status": 400,
-            "summary": "Invalid request body",
-            "description": error_as_dictionary,
-        }
+        error_response = _fail(
+            message=error_as_dictionary, status="Invalid request body"
+        )
+
+        return error_response
```

Comment:
I think we might have a type inconsistency here. Looks like the `message` argument of `_fail` is intended to be a string, but `error_as_dictionary` is a dict. Maybe we should just drop line 64 and pass `error` directly to `_fail`.

Reply:
Sounds good.  Will fix this.

Reply:
Oops actually looks like `error.json()` is a string. `error` is an `ValidationError` object.